File: android/src/main/kotlin/dev/eggnstone/chime/observers/ChimeMetricsObserver.kt
package dev.eggnstone.chime.observers

import com.amazonaws.services.chime.sdk.meetings.audiovideo.metric.MetricsObserver
import com.amazonaws.services.chime.sdk.meetings.audiovideo.metric.ObservableMetric
import io.flutter.plugin.common.EventChannel.EventSink
import org.json.JSONObject

class ChimeMetricsObserver(private val _eventSink: EventSink) : MetricsObserver
{
    override fun onMetricsReceived(metrics: Map<ObservableMetric, Any>)
    {
        val jsonObject = JSONObject()
        val eventArguments = JSONObject()
        eventArguments.put("Metrics", convertMetricsToJson(metrics))
        jsonObject.put("Name", "OnMetricsReceived")
        jsonObject.put("Arguments", eventArguments)
        _eventSink.success(jsonObject.toString())
    }

    private fun convertMetricsToJson(metrics: Map<ObservableMetric, Any>): JSONObject
    {
        val jsonObject = JSONObject()

        for (metric in metrics)
            jsonObject.put(metric.key.toString(), metric.value.toString())

        return jsonObject
    }
}

File: ios/Classes/observers/ChimeMetricsObserver.swift
//
//  ChimeMetricsObserver.swift
//  eggnstone_amazon_chime
//
//  Created by Hendrik Lakämper on 30.07.2020
//

import Foundation
import AmazonChimeSDK
import Flutter

public class ChimeMetricsObserver : MetricsObserver {
    let _eventSink: FlutterEventSink
    
    init(eventSink: @escaping FlutterEventSink) {
        self._eventSink = eventSink
    }
    
    public func metricsDidReceive(metrics: [AnyHashable : Any]) {
//        let json = """
//        {
//        "Name": "OnAudioSessionCancelledReconnect"
//        }
//        """
//        
//        _eventSink(json)
    }
    
//    func convertMetricsToJson(metrics: [AnyHashable : Any]) -> String {
//        return """
//        [
//        \(metrics.map({ (metric: [AnyHashable: Any]) -> String in
//            return """
//        \(metric.)
//        """
//
//        }))
//
//        ]
//        """
//    }
}
